src/lib/y2storage/disk_analyzer.rb

```diff
@@ -322,27 +289,15 @@ def find_mbr_gap
       gaps
     end
 
-    # Find any MS Windows partitions that could possibly be resized.
-    #
-    # This involves mounting any Windows-like partition to check if there are
-    # some typical directories (/windows/system32).
-    #
-    # Notice that by default this is not called if any Linux partitions were
-    # found on any of the candidate disks (i.e., on any disk except the
-    # installation medium). This can be called independently from the
-    # outside, though.
-    #
-    # @return [Array <::Storage::Partition>]
-    #
+    # @see #windows_partitions
     def find_windows_partitions
-      return [] unless Arch.x86_64 || Arch.i386
+      return {} unless windows_architecture?
```

Comment:
Yes. The old one was a bug. The method returns `Hash{String => Array<::Storage::Partition>}`